### PR TITLE
support GitHub enterprise

### PR DESCRIPTION
**Pull request by mashbourne**
Support for DANGER_GITHUB_API_BASE_URL env var http://danger.systems/guides/getting_started.html#enterprise-github

Resolves https://github.com/danger/danger-js/issues/153

### CODE REVIEW

File: source/platforms/github/GitHubAPI.ts [hunk 1]
```diff
@@ -129,8 +129,10 @@ export class GitHubAPI {
     if (this.token !== undefined) {
       headers["Authorization"] = `token ${this.token}`
     }
-
-    return this.fetch(`https://api.github.com/${path}`, {
+
```
Comment (macklinu) [inline source/platforms/github/GitHubAPI.ts]: Trailing whitespace

File: source/platforms/github/GitHubAPI.ts [hunk 2]
```diff
@@ -129,8 +129,10 @@ export class GitHubAPI {
     if (this.token !== undefined) {
       headers["Authorization"] = `token ${this.token}`
     }
-
-    return this.fetch(`https://api.github.com/${path}`, {
+    
+    const baseUrl = process.env["DANGER_GITHUB_API_BASE_URL"] || "https://api.github.com"
+
```
Comment (macklinu) [inline source/platforms/github/GitHubAPI.ts]: Trailing whitespace

**Comment by deecewan** [comment]
Thanks for the changes!

**Comment by alex3165** [comment]
When do you plan to release this ?

**Comment by orta** [comment]
very soon, just wanted to get #156 in and then do a bit of cleanup now

**Comment by orta** [comment]
shipped in 0.12.0

**Pull request merged**